packages/next/src/client/components/react-dev-overlay/server/middleware.ts

```diff
@@ -241,9 +284,30 @@ export async function getSource(
   for (const compilation of getCompilations()) {
     // TODO: `ignoreList`
     const sourceMap = await getSourceMapFromCompilation(moduleId, compilation)
+    const ignoreList = []
+    const moduleFilenames = sourceMap?.sources ?? []
+
+    // console.log('moduleFilenames', moduleFilenames)
+    for (let index = 0; index < moduleFilenames.length; index++) {
+      // bundlerFilePath case: webpack://./app/page.tsx
```

Comment:
question: Are you saying that every entry in `sources` always has a `webpack://` prefix? That would mean the `devtoolModuleFilenameTemplate` is ignored when `getSourceMapFromCompilation` is used.

Reply:
For one single page yes. What do you mean "every enrty", can you give a test example that there're multiple entries in sources?

Reply:
> can you give a test example that there're multiple entries in sources?

That is how source maps are specced. `sources` wouldn't need to be an array if there is only ever one entry. Sourcemaps with multiple sources occur when the generated generated code is constructed by concatenating multiple original source files.